cmd/remote-configuration/operator/client.go

```diff
@@ -0,0 +1,200 @@
+	// Set the defaults
+	collector.Default()
+	collector.TypeMeta.Kind = CollectorResource
+	collector.TypeMeta.APIVersion = v1alpha1.GroupVersion.String()
+	collector.ObjectMeta.Name = name
+	collector.ObjectMeta.Namespace = namespace
+
+	if collector.ObjectMeta.Labels == nil {
+		collector.ObjectMeta.Labels = map[string]string{}
+	}
+	collector.ObjectMeta.Labels[ResourceIdentifierKey] = ResourceIdentifierValue
+	err := collector.ValidateCreate()
+	if err != nil {
+		return err
+	}
+	c.log.Info("Creating collector")
+	return c.k8sClient.Create(ctx, collector)
+}
+
+func (c Client) update(ctx context.Context, old *v1alpha1.OpenTelemetryCollector, new *v1alpha1.OpenTelemetryCollector) error {
+	new.ObjectMeta = old.ObjectMeta
+	new.TypeMeta = old.TypeMeta
+	err := new.ValidateUpdate(old)
+	if err != nil {
+		return err
+	}
+	c.log.Info("Updating collector")
+	return c.k8sClient.Update(ctx, new)
+}
+
+func (c Client) Apply(name string, namespace string, configmap *protobufs.AgentConfigFile) error {
+	c.log.Info("Received new config", "name", name, "namespace", namespace)
+	var collectorSpec v1alpha1.OpenTelemetryCollectorSpec
+	err := yaml.Unmarshal(configmap.Body, &collectorSpec)
+	if err != nil {
+		return err
+	}
+	if len(collectorSpec.Config) == 0 {
+		return errors.NewBadRequest("Must supply valid configuration")
+	}
+	reasons, validateErr := c.validate(collectorSpec)
+	if validateErr != nil {
+		return validateErr
+	}
+	if len(reasons) > 0 {
+		return errors.NewBadRequest(fmt.Sprintf("Items in config are not allowed: %v", reasons))
+	}
+	collector := &v1alpha1.OpenTelemetryCollector{Spec: collectorSpec}
+	ctx := context.Background()
+	instance, err := c.GetInstance(name, namespace)
+	if err != nil {
+		return err
+	}
+	if instance != nil {
+		return c.update(ctx, instance, collector)
+	}
+	return c.create(ctx, name, namespace, collector)
+}
+
+func (c Client) Delete(name string, namespace string) error {
+	ctx := context.Background()
+	result := v1alpha1.OpenTelemetryCollector{}
+	err := c.k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}, &result)
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	return c.k8sClient.Delete(ctx, &result)
+}
+
+func (c Client) ListInstances() ([]v1alpha1.OpenTelemetryCollector, error) {
+	ctx := context.Background()
+	result := v1alpha1.OpenTelemetryCollectorList{}
+	err := c.k8sClient.List(ctx, &result, client.MatchingLabels{
+		ResourceIdentifierKey: ResourceIdentifierValue,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return result.Items, nil
+}
+
+func (c Client) GetInstance(name string, namespace string) (*v1alpha1.OpenTelemetryCollector, error) {
+	ctx := context.Background()
+	result := v1alpha1.OpenTelemetryCollector{}
+	err := c.k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}, &result)
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Would it make sense to do this?
```suggestion
	if err != nil {
		if errors.IsNotFound(err) ́{
			return nil, nil
		}
		return nil, err
	}
```

Reply:
I usually try and not nest `if` blocks where possible, but your suggestion may read better. Thank you for the suggestion!